src/_imaging.c

```diff
@@ -1571,7 +1571,22 @@ if (PySequence_Check(op)) { \
                 PyErr_SetString(PyExc_TypeError, must_be_sequence);
                 return NULL;
             }
-            int endian = strncmp(image->mode, "I;16", 4) == 0 ? (strcmp(image->mode, "I;16B") == 0 ? 2 : 1) : 0;
+            // 0 = none, 1 = little, 2 = big
+            int endian = 0;
+            if (strncmp(image->mode, "I;16", 4) == 0) {
+                if (strcmp(image->mode, "I;16") == 0 || strcmp(image->mode, "I;16L") == 0) {
+                    endian = 1;
+                } else if (strcmp(image->mode, "I;16B") == 0) {
+                    endian = 2;
+                } else {
+                    // native endianness
+#ifdef WORDS_BIGENDIAN
+                    endian = 2;
+#else
+                    endian = 1;
+#endif
```

Comment:
```suggestion
                if (strcmp(image->mode, "I;16N") == 0) {
#ifdef WORDS_BIGENDIAN
                    endian = 2;
#else
                    endian = 1;
#endif
                } else if (strcmp(image->mode, "I;16B") == 0) {
                    endian = 2;
                } else {
                    endian = 1;
                }
```
This seems simpler to me?